packages/rpc/src/vatidation/validator.ts

```diff
@@ -203,6 +208,27 @@ export class UnsafeValidator implements IValidator {
                 this.usingTenderly
             ) as Promise<ExecutionResult>
         }
+
+        if (this.balanceStateOverrideAllowed) {
+            const error = await simulateHandleOp(
+                userOperation,
+                this.entryPoint,
+                this.publicClient,
+                false,
+                zeroAddress,
+                "0x"
+            )
+
+            if (error.result === "failed") {
+                throw new RpcError(
+                    `UserOperation reverted during simulation with reason: ${error.data}`,
+                    ExecutionErrors.UserOperationReverted
+                )
+            }
+
+            return error.data
```

Comment:
uh... does this have the correct return type?

Reply:
thankfully yes <3